src/Stratis.Bitcoin/Connection/PeerBanning.cs

```diff
@@ -120,16 +122,19 @@ public void BanPeer(IPEndPoint endpoint, string reason = null)
                 ConnectionManagerBehavior peerBehavior = peer.Behavior<ConnectionManagerBehavior>();
                 if (!peerBehavior.Whitelisted)
                 {
-                    banPeer = false;
                     this.logger.LogDebug("Peer '{0}' banned for reason '{1}'.", endpoint, reason ?? "unknown");
                     peer.DisconnectAsync($"The peer was banned, reason: {reason}");
                 }
-                else this.logger.LogTrace("Peer '{0}' is whitelisted, for reason '{1}' it was not banned!", endpoint, reason ?? "unknown");
+                else
+                {
+                    banPeer = false;
+                    this.logger.LogTrace("Peer '{0}' is whitelisted, for reason '{1}' it was not banned!", endpoint, reason ?? "unknown");
+                }
             }
 
             if (banPeer)
             {
-                this.banStore.BanPeer(endpoint, this.dateTimeProvider.GetUtcNow().AddSeconds(this.connectionManagerSettings.BanTimeSeconds));
+                this.banStore.BanPeer(endpoint, this.dateTimeProvider.GetUtcNow().AddSeconds(banTimeSeconds > 0 ? banTimeSeconds : this.connectionManagerSettings.BanTimeSeconds));
```

Comment:
hmm the logic is unclear here
in consensus loop you have explained in comments that those constants are special
but here you only default to 0, but you don't mention it is somehow special
so i'm confused, is it here -1 and 0 also special or not?

if yes then -1 here would actually default to BanTimeSeconds which is probably not intended
i think you should use the constants here as well not `= 0` and `> 0`

Reply:
Ok this actually raises the questiosn should we put this logic here at all.
Maybe just check and if  `banTimeSeconds== 0` then don't ban at all

Reply:
FYI on the interface there is an explanation on the behaviour of the zero.

Reply:
Ok as suggested I will move the constants in to the PeerBan class that will work for all cases.

Reply:
Hmm I am actually not so happy about moving the constants in to the banning class as they represents actions in consensus, I would rather leave them there and not have unban logic in the PeerBan class.

Reply:
on interface i see this:
`        /// <param name="banTimeSeconds">The time in seconds this peer should be banned.</param>`

so i don't see any special values there

Reply:
ok new version, checking ..